src/metabase/pulse/render/js_svg.clj

```diff
@@ -39,6 +39,18 @@
   ;; todo is this thread safe? Should we have a resource pool on top of this? Or create them fresh for each invocation
   (delay (static-viz-context)))
 
+(defn default-colors
+  []
+  (let [{:keys [default light dark]} (-> (js/execute-fn-name @context "default_colors")
```

Comment:
It feels to me since we can call JS from BE, it might be better to pass the whole `public-settings/application-colors` to JS code, and modify its interface to support that. Then we could just use the return value from JS directly without having to calculate any additional colors e.g. lights or dark on BE.

Now, FE has an interface for getting a single color, what we could change is to make `default_colors` (FE implementation) return the whole color palette, which you kind of have already done here, but instead of having BE do any additional work e.g. combining maps, you could have `default_colors` (FE implementation) define a list of all colors then incrementally build a complete map of default colors using `color()` function and pass `public-settings/application-colors` as an argument (that needs modification on FE)

I'm not sure if what I wrote makes sense here. But here's the diagram of what I've been talking about
```mermaid
flowchart TD

subgraph BE
Someone -->|calls| B[application-colors fn]
B --> |calls| E[default-colors]
end

subgraph FE static viz interface
E --> |pass `public-settings/application-colors`| C[default_colors fn]
C --> |repeatedly calls w/ all color names| D[color fn]
end
```

This seems more straightforward than having to have BE implement what FE has already implemented.

Since, `color()` already accept a second argument `palette`

https://github.com/metabase/metabase/blob/fa0a2898352a91d313f8b60fb641d878ef11e1c1/frontend/src/metabase/lib/colors/palette.ts#L83

You'd only need to change your implementation of `default_colors` to accept an argument that'll be passed from BE (`public-settings/application-colors`)

Let me know what you think about this.